x-pack/plugins/enterprise_search/public/applications/enterprise_search_content/components/search_index/components/header_actions/header_actions.tsx

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React from 'react';
+
+import { useActions, useValues } from 'kea';
+
+import {
+  EuiButton,
+  EuiPopover,
+  EuiContextMenuPanel,
+  EuiContextMenuItem,
+  EuiText,
+} from '@elastic/eui';
+
+import { i18n } from '@kbn/i18n';
+
+import { APP_SEARCH_PLUGIN } from '../../../../../../../common/constants';
+import { ENGINE_CREATION_PATH } from '../../../../../app_search/routes';
+import { KibanaLogic } from '../../../../../shared/kibana';
+
+import { HeaderActionsLogic } from './header_actions.logic';
+
+const SearchEnginesPopover: React.FC = () => {
+  const { isSearchEnginesPopoverOpen } = useValues(HeaderActionsLogic);
```

Comment:
question: do you need a logic file for this, or can this just be handled with `useState` local to the component?

Reply:
I don't, for this and other two as well. It is a habit that I have to move every data to redux. I had a use cases to open some certain UI elements with pre populated redux store before, that helped me there. Do we have any preference on our code base in general? I can change them to state.